src/Umbraco.Web.Common/ImageProviders/FileProviderImageProvider.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.FileProviders;
+using SixLabors.ImageSharp.Web;
+using SixLabors.ImageSharp.Web.Providers;
+using SixLabors.ImageSharp.Web.Resolvers;
+
+namespace Umbraco.Cms.Web.Common.ImageProviders
+{
+    /// <summary>
+    /// Returns images from an <see cref="IFileProvider"/> abstraction.
+    /// </summary>
+    public class FileProviderImageProvider : IImageProvider
```

Comment:
I like the pattern you've used here. You might be able to convince me to add this upstream.

Reply:
Would be great if you're willing to merge this into ImageSharp.Web, will create a PR for it 🎉 It can/should lose the virtual methods for better performance when not inherited and sealed, so you can also use this with a factory registration:
```c#
.AddProvider(provider => new FileProviderImageProvider(
    provider.GetRequiredService<IWebHostEnvironment>().WebRootFileProvider,
    provider.GetRequiredService<FormatUtilities>()))
```